bullet-featherstone/src/SimulationFeatures.cc

```diff
@@ -37,14 +37,15 @@ void SimulationFeatures::WorldForwardStep(
   const auto worldInfo = this->ReferenceInterface<WorldInfo>(_worldID);
   auto *dtDur =
     _u.Query<std::chrono::steady_clock::duration>();
+  double stepSize = 0.001;
   if (dtDur)
   {
     std::chrono::duration<double> dt = *dtDur;
     stepSize = dt.count();
   }
 
-  worldInfo->world->stepSimulation(static_cast<btScalar>(this->stepSize), 1,
-                                   static_cast<btScalar>(this->stepSize));
+  worldInfo->world->stepSimulation(static_cast<btScalar>(stepSize), 1,
```

Comment:
I'm not aware if the step size can change during runtime in gz-sim, but the API definitely allows it. I guess the proper way to implement this is to update all the motors in this function similar to https://github.com/bulletphysics/bullet3/blob/e9c461b0ace140d5c73972760781d94b7b5eee53/examples/Importers/ImportMJCFDemo/ImportMJCFSetup.cpp#L345-L363. I won't block on this, but maybe a TODO here would be good.

Reply:
Can you clarify this? Is the goal to make sure we apply target velocity on every step (instead of just once when user calls `SetJointVelocityCommand`).

Reply:
If `dt` is different than the step size configured in the SDF file, we'd have to update the max impulse since that requires multiplying the force with the step size. In the example I linked, they update all constraints during `step`. So, I'm just suggesting we put a TODO here to document that providing a different `dt` here won't work properly.

Reply:
Got it. I added todo note in [a992537](https://github.com/gazebosim/gz-physics/pull/658/commits/a992537f4062c5bd71ff31d3e507a37c7b5fa1cc)